demo/app/ios/Runner/flutterPlugin.swift

```diff
@@ -322,15 +322,48 @@ public class SwiftWalletSDKPlugin: NSObject, FlutterPlugin {
                                                      details: "Pass scopes, clientID and redirectURI as the arguments"))
                 }
                 
+                let dynamicRegistrationSupported = try openID4CI.dynamicRegistrationSupported()
+                var clientID = authCodeArgs["clientID"]
+                let redirectURI = authCodeArgs["redirectURI"]! as? String
+                let scopes = authCodeArgs["scopes"]! as! [String]
+                
+                if (dynamicRegistrationSupported.boolValue == true) {
+                    let dynamicRegistrationEndpoint = try openID4CI.dynamicRegistrationEndpoint()
+                    let clientMetadata = Oauth2ClientMetadata()
+                    let grantTypesArr = ApiStringArray()
+                    grantTypesArr?.append("authorization_code")
+                    clientMetadata?.setGrantTypes(grantTypesArr)
+                
+                    
+                    let redirectUri = ApiStringArray()
+                    redirectUri?.append(redirectURI)
+                    clientMetadata?.setRedirectURIs(redirectUri)
+            
+                
+                    let scopesUpdated = scopes.joined(separator: " ")
+                    clientMetadata?.setScope(scopesUpdated)
+                    clientMetadata?.setTokenEndpointAuthMethod("none")
+                    
+                    let authorizationCodeGrantParams = try openID4CI.getAuthorizationCodeGrantParams()
+                    
+                    if authorizationCodeGrantParams.hasIssuerState() {
+                        let issuerState = authorizationCodeGrantParams.issuerState(nil)
+                        clientMetadata?.setIssuerState(issuerState)
+                    }
+                    
+                    let registrationResp = Oauth2RegisterClient(dynamicRegistrationEndpoint, clientMetadata, nil, nil)
+                    clientID = (registrationResp?.clientID())!
+                }
+                
                 if (!authCodeArgs.keys.contains("scopes")) {
-                    authorizationLink = try openID4CI.createAuthorizationURL(clientID: authCodeArgs["clientID"]! as! String, redirectURI: authCodeArgs["redirectURI"]! as! String)
+                    authorizationLink = try openID4CI.createAuthorizationURL(clientID: clientID as! String, redirectURI: redirectURI as! String)
                 } else {
-                    authorizationLink = try openID4CI.createAuthorizationURLWithScopes(scopes: authCodeArgs["scopes"]! as! [String], clientID: authCodeArgs["clientID"]! as! String, redirectURI: authCodeArgs["redirectURI"]! as! String)
+                    authorizationLink = try openID4CI.createAuthorizationURLWithScopes(scopes: scopes as! [String], clientID: clientID as! String, redirectURI: redirectURI as! String)
```

Comment:
I see okay will update this

Reply:
@DRK3 I recall now the only reason I didn't do this was due to this string array and I was not able to set the string array to string back that conversion was failing , so since you mentioned u will be pushing soon so its just one line change I will change it. Right now if I try to do there are warning and compile errors.

Reply:
Sure, after I make the change to Wallet-SDK I can update this to pass the scopes through